### PR TITLE
Sql alchemy targets

### PR DESCRIPTION
**Pull request by gouthambs**
Provides the feature described in #592

### CODE REVIEW

**Comment by gouthambs** [comment]
@erikbern what would be a good place to document this addition? I haven't had a chance to do that yet.

**Comment by erikbern** [comment]
I think you can put documentation in the code. It will be auto-generated and uploaded to readthedocs.org

File: luigi/contrib/sqla.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+from sqlalchemy import Table, MetaData, Column, String, DateTime, create_engine, select
+
+logger = logging.getLogger('luigi-interface')
+
+
+class SQLAlchemyTarget(luigi.Target):
+    """Target for a resource in database using SQLAlchemy.
+
+    This will rarely have to be directly instantiated by the user"""
+    marker_table = luigi.configuration.get_config().get('sqlalchemy', 'marker-table', 'table_updates')
```
Comment (erikbern) [inline luigi/contrib/sqla.py]: I would prefer it not to hit configuration in the global scope. Can you set it `None` and look it up when needed?

**Comment by gouthambs** [comment]
@erikbern I have removed the configuration usage in the global scope. Take a look and let me know if you have any more thoughts.

**Comment by erikbern** [comment]
LGTM – I'll give it a few days in case anyone else has thoughts (@freider @Tarrasch @daveFNbuck maybe?)

**Comment by Tarrasch** [comment]
Hi @gouthambs , I would like to hear if you have/are used this in production. What motivated you to do this? Also I want to give +100 on the documentation! On a glance, this looks very well written!

**Comment by gouthambs** [comment]
@Tarrasch I am employing this part of a development project that I am woking on. It is not in production yet.

I use SQLAlchemy in my projects which helps me to not worry so much about the database layer. My motivation to write the SQLAlchemyTarget was to have the same freedom in some of my processing jobs. Though this is the first time I am working with luigi.

**Comment by gouthambs** [comment]
@erikbern The docs are not showing up for this module.

**Comment by erikbern** [comment]
Yeah that's pretty weird. Looks like readthedocs rebuilt the docs but didn't include this submodule: http://luigi.readthedocs.org/en/stable/

Anyone has an idea?

**Comment by gouthambs** [comment]
Should there be an entry in doc/api/luigi.contrib.rst?

**Comment by erikbern** [comment]
I think that explains it yeah. Would you mind adding it?

**Comment by gouthambs** [comment]
@erikbern #596 adds this.

**Pull request merged**